### ISSUE TITLE
Big memory leak when reimporting images(~ 33MB ~ 50 reimports)

### ISSUE THREAD

**Issue by qarmin** [godotengine/godot#29741]
**Godot version:**
3.2.dev.custom_build. 15425b450

**OS/device including version:**
Ubuntu 19.04

**Issue description:**
When I on empty scene, just reimport Godot basic image(64x64), then really big leak happens. It is easy too see in system monitor.
```
48,216 bytes in 123 blocks are definitely lost in loss record 151 of 163
   at 0x584574F: malloc (in /usr/lib/x86_64-linux-gnu/valgrind/vgpreload_memcheck-amd64-linux.so)
   by 0x60AACE2: png_create_info_struct (in /usr/lib/x86_64-linux-gnu/libpng16.so.16.36.0)
   by 0x1F1EBC0: _lossless_pack_png(Ref<Image> const&) (image_loader_png.cpp:295)
   by 0x234A2E8: ResourceImporterTexture::_save_stex(Ref<Image> const&, String const&, int, float, Image::CompressMode, bool, int, bool, bool, bool, bool, bool, bool, bool) (resource_importer_texture.cpp:285)
   by 0x234BD92: ResourceImporterTexture::import(String const&, String const&, Map<StringName, Variant, Comparator<StringName>, DefaultAllocator> const&, List<String, DefaultAllocator>*, List<String, DefaultAllocator>*, Variant*) (resource_importer_texture.cpp:531)
   by 0x1F9F8AA: EditorFileSystem::_reimport_file(String const&) (editor_file_system.cpp:1789)
   by 0x1FA236D: EditorFileSystem::reimport_files(Vector<String> const&) (editor_file_system.cpp:1987)
   by 0x2152DDB: ImportDock::_reimport() (import_dock.cpp:476)
   by 0x215235E: ImportDock::_reimport_attempt() (import_dock.cpp:421)
   by 0x142BDDE: MethodBind0::call(Object*, Variant const**, int, Variant::CallError&) (method_bind.gen.inc:59)
   by 0x3687E00: Object::call(StringName const&, Variant const**, int, Variant::CallError&) (object.cpp:940)
   by 0x3689B1C: Object::emit_signal(StringName const&, Variant const**, int) (object.cpp:1240)

16,504,588 (351,168 direct, 16,153,420 indirect) bytes in 59 blocks are definitely lost in loss record 162 of 163
   at 0x584574F: malloc (in /usr/lib/x86_64-linux-gnu/valgrind/vgpreload_memcheck-amd64-linux.so)
   by 0x60B0E73: png_malloc_warn (in /usr/lib/x86_64-linux-gnu/libpng16.so.16.36.0)
   by 0x622E384: deflateInit2_ (in /usr/lib/x86_64-linux-gnu/libz.so.1.2.11)
   by 0x60CA098: ??? (in /usr/lib/x86_64-linux-gnu/libpng16.so.16.36.0)
   by 0x60CC632: ??? (in /usr/lib/x86_64-linux-gnu/libpng16.so.16.36.0)
   by 0x60CD17D: ??? (in /usr/lib/x86_64-linux-gnu/libpng16.so.16.36.0)
   by 0x60C7ABC: png_write_row (in /usr/lib/x86_64-linux-gnu/libpng16.so.16.36.0)
   by 0x60C7E17: png_write_image (in /usr/lib/x86_64-linux-gnu/libpng16.so.16.36.0)
   by 0x1F1EFF0: _lossless_pack_png(Ref<Image> const&) (image_loader_png.cpp:376)
   by 0x234A2E8: ResourceImporterTexture::_save_stex(Ref<Image> const&, String const&, int, float, Image::CompressMode, bool, int, bool, bool, bool, bool, bool, bool, bool) (resource_importer_texture.cpp:285)
   by 0x234BD92: ResourceImporterTexture::import(String const&, String const&, Map<StringName, Variant, Comparator<StringName>, DefaultAllocator> const&, List<String, DefaultAllocator>*, List<String, DefaultAllocator>*, Variant*) (resource_importer_texture.cpp:531)
   by 0x1F9F8AA: EditorFileSystem::_reimport_file(String const&) (editor_file_system.cpp:1789)

17,307,904 (83,456 direct, 17,224,448 indirect) bytes in 64 blocks are definitely lost in loss record 163 of 163
   at 0x584574F: malloc (in /usr/lib/x86_64-linux-gnu/valgrind/vgpreload_memcheck-amd64-linux.so)
   by 0x60B0E73: png_malloc_warn (in /usr/lib/x86_64-linux-gnu/libpng16.so.16.36.0)
   by 0x60AAC40: ??? (in /usr/lib/x86_64-linux-gnu/libpng16.so.16.36.0)
   by 0x60C788D: png_create_write_struct_2 (in /usr/lib/x86_64-linux-gnu/libpng16.so.16.36.0)
   by 0x60C7930: png_create_write_struct (in /usr/lib/x86_64-linux-gnu/libpng16.so.16.36.0)
   by 0x1F1EB5E: _lossless_pack_png(Ref<Image> const&) (image_loader_png.cpp:291)
   by 0x234A2E8: ResourceImporterTexture::_save_stex(Ref<Image> const&, String const&, int, float, Image::CompressMode, bool, int, bool, bool, bool, bool, bool, bool, bool) (resource_importer_texture.cpp:285)
   by 0x234BD92: ResourceImporterTexture::import(String const&, String const&, Map<StringName, Variant, Comparator<StringName>, DefaultAllocator> const&, List<String, DefaultAllocator>*, List<String, DefaultAllocator>*, Variant*) (resource_importer_texture.cpp:531)
   by 0x1F9F8AA: EditorFileSystem::_reimport_file(String const&) (editor_file_system.cpp:1789)
   by 0x1FA236D: EditorFileSystem::reimport_files(Vector<String> const&) (editor_file_system.cpp:1987)
   by 0x2152DDB: ImportDock::_reimport() (import_dock.cpp:476)
   by 0x215235E: ImportDock::_reimport_attempt() (import_dock.cpp:421)
```

**Comment by Phoenix1747** [comment]
Can somewhat confirm. Reimporting just the standard icon.png Godot is using 0.1 - 0.2 MB more of memory on my Windows machine everytime I click the reimport button. Not as drastic as on your Ubuntu machine though 🤔

**Comment by akien-mga** [comment]
Fixed by #29874.
